services/ingestor/src/main/kotlin/company/thewebhook/ingestor/Application.kt

```diff
@@ -47,16 +50,28 @@ fun Application.module() = launch {
             .lowercase()
             .split(",")
 
+    val serverHostRegexString =
+        environment.config.propertyOrNull("ingestor.validation.serverHostRegex")?.getString()
+            ?: throw ServerHostRegexException("Serverhost regex string not provided")
+
     val producer by inject<Producer<ByteArray>>()
     producer.connect(messageStoreConfig)
 
+    val serverHostRegex = Regex(serverHostRegexString)
+
     routing {
         HttpMethod.DefaultMethods.forEach {
             route("{...}", it) {
                 handle {
-                    val serverHost = call.request.origin.serverHost
+                    val serverHost = call.request.origin.serverHost.lowercase()
                     val remoteHost = call.request.origin.remoteHost
```

Comment:
let's do lowercase here too for symmetry and possible future validations/usages.

Reply:
Sure